### PR TITLE
added script for spelling checker

### LINKED ISSUE #948: Script that automatic Spell-checking / correcting for a given input file/text
# Description

A script to implement spell checking and rectifying for a given input text in PDF form, Word Doc or other popular format. A great feature that can be used for automated software that does answer-sheet checking, Resume anaylsis or a tool to improve language.

## Language

- [X] Go
- [X] Javascript
- [X] Python

## Checklist:

 

- **Details on the Script**

| name            | about                                                  | labels  | assignees                                                      |
| -------------------------------------------------------------- | -------------------------------------------------------------- | -------------------------------------------------------------- | -------------------------------------------------------------- |
| Spell-checker | Script to perform a spell checking/ correcting for the text | JavaScript, Python, good-first-issue, medium |  |

**Describe the solution you'd like**: 
Spell checking tools are a quite helpful for various tasks especially in language-based applications. This script can provide much value for implementing such feature.

**Additional context**: 

The must-have features: Spell Checking, logging the mistaken words etc.
Good to have features: Grammatical mistakes also checked and suggests changes.

Not needed to change those words in the file. As the first iteration, just logging/telling the use where one can improve and what to improve can be a good start. Make sure the line numbers are logged along as well. There are various `npm` packages and python Libs available for doing this. 

Though, it is not limited to this. Would love to discuss how and what can be done. Feel free to ping me re: this feature implementation.

### PR DESCRIPTION
**Pull request by tanishq-arya**
# Description
Added the script for spelling checker.
Fixes #948 

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ x] This change requires a documentation update
- [ ] Documentation Update

# Checklist:

- [ ] My code follows the style guidelines(Clean Code) of this project
- [ ] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works
Screenshot (1242)

### CODE REVIEW

**Comment by tanishq-arya** [comment]
@sohamsshah Please review this PR. 
I will add the README soon. 
Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.

**Comment by vybhav72954** [comment]
> @sohamsshah Please review this PR.
> I will add the README soon.
> Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.

Which exact module are we talking about here?

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    if user_answer.lower()=='y' or user_answer.lower()=='yes':
+        # call to correctErrors function
+        correctErrors(errorList)
+        print("Check the output.txt file for the corrected text.")
+        print("Thankyou for using spelling checker program.")
+    else:
+        print("Thankyou for using spelling checker program.")
+ 
+# call to the main function
+main()
```
Comment (vybhav72954) [inline Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:85]: Pls add a blank line here

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from nltk import word_tokenize
+
+# create an instance of the spellchecker
+spell = SpellChecker()
+
+# tokens --> stores the tokenized words
+tokens = []
+
+# function to read the input file
+def readTextFile(textFilename):
```
Comment (vybhav72954) [inline Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:13]: Add a docstring please

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+    # Create a list of words from these tokens checking if the word is alphanumeric 
+    words = [
+        word 
+        for word in tokens if word.isalpha()
+    ]
+    inputFile.close()
+    return words
+
+# function to detect errors in the file if any
+def findErrors(textWords):
```
Comment (vybhav72954) [inline Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:28]: Add a docstring please

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    misspelledWords = []
+    for word in textWords:
+        # correction() --> method of spellchecker module to correct the word
+        if spell.correction(word) != word:
+            misspelledWords.append(word)
+
+    return misspelledWords
+
+# function to print errors
+def printErrors(errorList):
```
Comment (vybhav72954) [inline Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:38]: Add a docstring please

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    print("Misspelled words are:")
+    for word in errorList:
+        # candidates() --> method of spellchecker module to find suitable corrections of the word
+        print(f'{word} : {spell.candidates(word)}')
+
+# function to correct errors
+def correctErrors(errorList):
```
Comment (vybhav72954) [inline Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:45]: Add a docstring please

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+            # if word is incorrect we replace it with the corrected word
+            word = spell.correction(word)
+        
+        # this writes text to the new output.txt file
+        outputFile.write(" ".join(tokens))
+
+    outputFile.close()
+
+# main function
+def main():
```
Comment (vybhav72954) [inline Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:59]: Add a docstring, please

And for every function remove the comment on the line before the defenition

**Review by vybhav72954 — CHANGES_REQUESTED** [review]

**Comment by tanishq-arya** [comment]
> > @sohamsshah Please review this PR.
> > I will add the README soon.
> > Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.
> 
> Which exact module are we talking about here?

spellchecker

File: Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py (hunk 6 above)
Comment (tanishq-arya) [reply Python/Automatic_Spelling_Checker_Corrector/automatic_spelling_checker_corrector.py:59]: okay

**Comment by vybhav72954** [comment]
> > > @sohamsshah Please review this PR.
> > > I will add the README soon.
> > > Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.
> > 
> > 
> > Which exact module are we talking about here?
> 
> spellchecker

Just to confirm yet again, `spellchecker` not `pyspellchecker` right?

**Comment by tanishq-arya** [comment]
> > > > @sohamsshah Please review this PR.
> > > > I will add the README soon.
> > > > Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.
> > > 
> > > 
> > > Which exact module are we talking about here?
> > 
> > 
> > spellchecker
> 
> Just to confirm yet again, `spellchecker` not `pyspellchecker` right?

Sorry my mistake its pyspellchecker

**Comment by vybhav72954** [comment]
> > > > > @sohamsshah Please review this PR.
> > > > > I will add the README soon.
> > > > > Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.
> > > > 
> > > > 
> > > > Which exact module are we talking about here?
> > > 
> > > 
> > > spellchecker
> > 
> > 
> > Just to confirm yet again, `spellchecker` not `pyspellchecker` right?
> 
> Sorry my mistake its pyspellchecker



> > > > > @sohamsshah Please review this PR.
> > > > > I will add the README soon.
> > > > > Also can you help me with the requirements file. When I performed the pip install in this directory I have an error with module not found. But it works on my local machine in a different directory when I installed.
> > > > 
> > > > 
> > > > Which exact module are we talking about here?
> > > 
> > > 
> > > spellchecker
> > 
> > 
> > Just to confirm yet again, `spellchecker` not `pyspellchecker` right?
> 
> Sorry my mistake its pyspellchecker

Was the pip install successful in the directory?

**Comment by tanishq-arya** [comment]
It is working now I installed it again. Will push the changes soon.

**Comment by tanishq-arya** [comment]
Should I add print statements to make this look more presentable like this?
Screenshot (1294)
Screenshot (1293)

**Comment by sohamsshah** [comment]
> Should I add print statements to make this look more presentable like this?
> Screenshot (1294)
> Screenshot (1293)

This looks wow! Yes, ofc any sort of improvement that can help the users/devs is appreciated.

**Comment by tanishq-arya** [comment]
Please let me know if the readme needs any changes.

**Review by sohamsshah — APPROVED** [review]
LGTM @HarshCasper

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**